### PR TITLE
fix krun script to use module specified by module attribute for kparse

### PR DESCRIPTION
**Pull request by dwightguth**
by default

Currently, if you specify the`parser` attribute for a configuration variable, krun will not actually store that module and use it to perform parsing, which leads to a performance regression because we use the jit parser even though a bison parser exists. The solution is simply to record the module we are supposed to use for parsing and use it by default when we call kparse.

### CODE REVIEW

File: k-distribution/src/main/scripts/bin/krun
Comment (ehildenb) [inline k-distribution/src/main/scripts/bin/krun:491]: AFAICT, this check will never pass, because we're always defining `indirectModuleName` above. But I also don't understand the `${!...}` syntax..

**Comment by radumereuta** [comment]
lgtm. I'll let Everett approve it when he's satisfied with his questions.
I see that I also added one such line, but I think I copy/pasted another instance and didn't pay attention to `!`.

File: k-distribution/src/main/scripts/bin/krun
Comment (dwightguth) [reply k-distribution/src/main/scripts/bin/krun:491]: It's an indirect variable lookup. We're not looking up indirectModuleName, we're expanding it as a string and looking up the variable with the name its value expands to. That variable might not be set if the user didn't specify the parser attribute for a particular configuration variable.

**Review by ehildenb — APPROVED** [review]

**Pull request merged**
